R/add_ppa_flag.R

```diff
@@ -0,0 +1,171 @@
+#' Add Potentially Preventable Admissions Marker
+#'
+#' @description This function takes a data frame input and determines, based on
+#' a combination of diagnostic codes and operation codes, whether an admission was
+#' preventable or not
+#' @param data A data frame
+#'
+#' @return A data frame to use as a lookup of PPAs
+#' @export
+add_ppa_flag <- function(data) {
+  expected_vars <- c(
+    "anon_chi", "cij_marker", "cij_pattype", "recid",
+    "op1a", "diag1", "diag2", "diag3", "diag4",
+    "diag5", "diag6"
+  )
+  if (!all(expected_vars %in% names(data))) {
+    missing_vars <- setdiff(expected_vars, names(data))
+
+    cli::cli_abort("{missing_vars} were missing from the data and are needed
+                     to assign preventable admissions")
+  }
```

Comment:
We'll use `check_variables_exist` from #387 here when that has been merged

Reply:
Added use of this function and altered tests to reflect this